test/e2e/operator_expectations.go

```diff
@@ -121,6 +115,10 @@ func (tc *testConfig) ExpectAllMachinesLinkedToANode() error {
 		nodeNameToMachineAnnotation[node.Name] = node.Annotations[machineAnnotationKey]
 	}
 	for _, machine := range machineList.Items {
```

Comment:
Consider putting the test under the `wait.PollImmediate` as well. It may take a while to set the `NodeRef` by the nodelink controller.